### PR TITLE
Resolve each training epoch resulting in progressively faster output

### PR DESCRIPTION
**Pull request by dragonfyre13**
Unsure if this part on input_embeddings was prepared for sub-talker (the non-autoregressive bit) originally, or some other reason why this might be here. Problem is, for the main autoregressive model (model.talker, which determines pacing, duration, prosody, etc. and is used for predicting codec_0_label) having it breaks causality. The AR model should be learning to look at text/codec layer 0, not including looking at layers 1-15, since these don't exist yet at actual inference time. Basically, instead of learning how to time speech based on text, it appears to be learning to time speech based on... the time speech took. End result, it's "rushing" through the generated audio, seemingly faster and faster with each training epoch performed. Remove this bit of code, and all that goes away. It doesn't even appear to have a distinct impact on loss reduction speed based on my testing with a few data sets (elise, jenny, several of my own used for other models).

### CODE REVIEW

**Comment by humblenginr** [comment]
hey @dragonfyre13 , did you also notice that the script does double-shifting to the labels? The forward function uses (https://github.com/QwenLM/Qwen3-TTS/blob/main/qwen_tts/core/models/modeling_qwen3_tts.py#L1731) uses the HF PreTrainedModel class's default loss_function, which is ForCausalLMLoss (https://github.com/huggingface/transformers/blob/main/src/transformers/loss/loss_utils.py#L57) which does internal shifting of labels. And the finetuning data preparation script also does label shifting.

**Comment by MrMuhannadObeidat** [comment]
I tried the new repo above from rekunenkdr, it is producing a very decent model so far. I only tested with 10K samples and even on the second epoch I am getting good arabic voice and no speed up. 
I will run a bigger test with 60K samples next.
One question, the model produced is now labeled as custom_voice instead of base. This means that creating a voice prompt will not be supported. How can I fix that?

**Comment by rekuenkdr** [comment]
> I tried the new repo above from rekunenkdr, it is producing a very decent model so far. I only tested with 10K samples and even on the second epoch I am getting good arabic voice and no speed up. I will run a bigger test with 60K samples next. One question, the model produced is now labeled as custom_voice instead of base. This means that creating a voice prompt will not be supported. How can I fix that?

Hi, are you using my streaming fork? 
Is not intented for finetuning, (although it should work for that too), but for streaming, I applied several patches, on that direction for a realtime AI assistant I'm developing, and fixed the speed issue for voice clone (as my streaming methods didn't have repetition_penalty at the beggining), not finetune.

I merged this PR just to try it whenever I have the time, but is still untested from my end.

You can try this: https://github.com/QwenLM/Qwen3-TTS/issues/179#issuecomment-3870059313
It seems to fix speed issues with finetune models too, we are all investigating this bug and testing.

All finetune models are labeled custom_voice to differenciate from base, but in essence is the base model, you can still create voice_prompt from them.

Load your checkpoint as you would load base and you are good to go.

**Comment by duarteocarmo** [comment]
Not sure why this is not merged yet - it fixed my 0.6B training getting insanely fast :)
